golang/cosmos/x/swingset/keeper/keeper.go

```diff
@@ -132,11 +159,49 @@ func (k Keeper) ActionQueueLength(ctx sdk.Context) (int32, error) {
 	if err != nil {
 		return 0, err
 	}
-	size := tail - head
-	if size > math.MaxInt32 {
+	size := tail.Sub(head)
+	if !size.IsInt64() {
+		return 0, fmt.Errorf("%s size too big: %s", StoragePathActionQueue, size)
+	}
+
+	int64Size := size.Int64()
+	if int64Size > math.MaxInt32 {
 		return math.MaxInt32, nil
 	}
-	return int32(size), nil
+	return int32(int64Size), nil
+}
+
+func (k Keeper) UpdateQueueAllowed(ctx sdk.Context) error {
+	params := k.GetParams(ctx)
+	inboundQueueMax, found := types.QueueSizeEntry(params.QueueMax, types.QueueInbound)
+	if !found {
+		return errors.New("could not find max inboundQueue size in params")
+	}
+	inboundMempoolQueueMax := inboundQueueMax / 2
```

Comment:
This is a separate entry: `types.QueueInboundMempool`. Please correct any comment or doc which says it's always half of `QueueInbound`.

Reply:
I don't understand, which docs or comment says that?

Reply:
That's what the existing behavior that I'm porting from JS is.

While we have a  `params.QueueMax` value for `QueueInbound`, we don't have an explicit param value for `QueueInboundMempool`, which is hardcoded and documented as 50% of the `QueueInbound` value.

However we do need to compute a a value for the `QueueInboundMempool` of the `state.QueueAllowed`. That value is indeed only going to be 50% of the allowed `QueueInbound` if the actionQueue is empty. However I don't believe there is any claim anywhere that the allowed `QueueInboundMempool` is always 50% of the allowed `QueueInbound`.